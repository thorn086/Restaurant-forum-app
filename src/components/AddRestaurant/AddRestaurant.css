.restaurant-form{
    border: 1px solid black;
    width: 90%;   
    background-color: rgb(70, 70, 193);
    margin:1rem;
    border-radius: 5%;
    box-shadow: 6px 6px rgb(61, 61, 95);
    padding: 1rem 1rem 1.5rem 1rem;
    text-align: center;
    height: auto;
    letter-spacing: 2px;
    color: white;
    font-size: 16px;
    font-weight: 600;
    line-height: 2;
    }
.restaurant_field{
    margin:2rem 0;
}
.error_field{
    background-color: rgb(53, 53, 53);
    color: red;
    text-align: left;
    font-size: 12px;
    padding: 1rem;
    border: 2px solid white;
    border-radius: 2rem;
    text-shadow: 1px 1px black;
}
   
.form{
        margin: .5rem .1rem;
    }
#state{
    background-color: white;
    text-align: center;
}

#state,#resName, #address, #phone,#comments{
    padding-left: 5px;
    border: 3px solid black;
    font-size: 14px;
    font-weight: 500;
    box-shadow: -1px -1px white;
}

.resForm, #author{
    margin: 1rem;
}
#resName, #address, #phone, #state{
    width: 15rem;
    height: 2rem;
}
#comments{
    height: 6rem;
    width: 15rem;
    padding: 1rem;
}
.back-btn.resForm{
    margin: auto;
}
@media (min-width: 426px){
    .restaurant-form{
        width: 80%;
        margin:auto;
    }
    #resName, #address, #phone, #state{
        width: 20rem;
        height: 2rem;
    }
    #comments{
        height: 6rem;
        width: 20rem;
        padding: 1rem;
    }
  
}
@media (min-width: 603px){
    .restaurant-form{
        width: 75%;
        margin:auto;
    }
    #resName, #address, #phone, #state{
        width: 20rem;
        height: 2rem;
    }
    #comments{
        height: 6rem;
        width: 20rem;
        padding: 1rem;
    }
  
}
@media (min-width: 699px){
    .restaurant-form{
        width: 95%;
        margin:auto;
    }
    .restaurant_field{
        padding-left: 10px;
        text-align: left;
    }
    .error_field{
        font-size: 16px;
        padding-left: 5rem;
    }
    #resName, #address, #phone, #state{
        width: 23rem;
        height: 2rem;
    }
     #phone{
        margin-left: 3rem;
    }
    #resName{
        margin-left: 3.3rem;
    }
    #state{
        margin-left: 3.5rem;
    }

    #address{
        margin-left: 2rem;
    }
    #comments{
        height: 6rem;
        width: 23rem;
        padding: 1rem;
        margin-left: 7.1rem;
        
    }
    .comments_rest{
       position: relative;
       top:-80px;
        
    }
 
}