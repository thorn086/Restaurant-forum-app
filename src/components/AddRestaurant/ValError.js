

 export default function validate(name,address,comments){
    const errors =[]
    if(name.length < 1 ){
        errors.push('Name must be longer than 1 and less than 25')
    }
    if(address.length < 1 || address.length <15){
        errors.push('You must enter an address')
    }
    if(comments.length === 0 || comments.length <15){
        errors.push('You must enter at least 15 characters of information')
    }
    return errors;
}
