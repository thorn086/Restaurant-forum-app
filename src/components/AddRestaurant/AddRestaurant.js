import React from 'react';
import config from '../../config';
import { withRouter } from 'react-router-dom';
import TokenService from '../../services/token-services';
import restForumContext from '../../context';
import './AddRestaurant.css';
import validate from './ValError';
class AddRestaurant extends React.Component {
    static defaultProps = {
        history: {
            push: () => {}
        },
        match: {
            params: {}
        }
    }
    static contextType = restForumContext

    state={   
    errors: []
    }
    handleSubmit = event => {
        event.preventDefault()
        const { id } = this.props.match.params;
        const stateName=event.target['restaurant-state'].value.split(' ');
        const name =event.target['restaurant-name'].value;
        const address= event.target['restaurant-address'].value;
        const phone= event.target['restaurant-phone'].value;
        const comments=event.target['restaurant-comments'].value;
        
        const errors = validate(name,address,comments);
        if(errors.length > 0){
            this.setState({errors})
            return;
        }

        const newRestaurant = {
            name,
            address,
            phone,
            state_id: parseInt(stateName[0]),
            city_id: id,
            comments,
        };

        fetch(`${config.API_ENDPOINT}/city/${id}`, {
            method: 'POST',
            headers: {
                'content-type': 'application/json',
                'authorization': `bearer ${TokenService.getAuthToken()}`
            },
            body: JSON.stringify(newRestaurant),
        })
            .then(results => {
                if (!results.ok)
                    return results.json().then(e => Promise.reject(e));
                return results.json();
            })
            .then(restaurant => {
                this.context.addRestaurant(restaurant);
                window.location.reload();
            })
            .catch(error => {
                console.error({ error });
            })
    }


    render() {
        const {errors} = this.state
        const valErrors = errors.map(error => (<p key={error}><span className='error-title'>Error:</span>{error}</p>))

        return (
            
            <form className="restaurant-form" onSubmit={this.handleSubmit}>
                <div className='error_field' style= {{display: (errors < 1) ? 'none':'block'}} >{valErrors}</div>
                <div className='restaurant_field'>
                    <label htmlFor='restaurant-name'>Restaurant Name</label>
                    <input type='text' id='restaurant-name' name='restaurant-name'placeholder='Enter Restaurant Name'></input>
                </div>
                <div className='restaurant_field'>
                    <label htmlFor='restaurant-address'>Restaurant Address</label>
                    <input type='text' id='restaurant-address' name='restaurant-address' placeholder='Enter Restaurant Address' ></input>
                </div>
                <div className='restaurant_field'>
                    <label htmlFor='restaurant-phone'>Restaurant Phone</label>
                    <input type='tel' id='restaurant-phone' pattern="^(?:\(\d{3}\)|\d{3})[- ]?\d{3}[- ]?\d{4}$" x-autocompletetype="tel" name='restaurant-phone' placeholder='555-555-5555' autofocus></input>
                </div>
                <div className='restaurant_field'>
                    <label htmlFor='restaurant-state'>Restaurant State</label>
                    <select type='text' id='restaurant-state' name='restaurant-state' >
                        <option>0 Select a state</option>
                        <option>1 Alabama</option>
                        <option>2 Alaska</option>
                        <option>3 Arizona</option>
                        <option>4 Arkansas</option>
                        <option>5 California</option>
                        <option>6 Colorado</option>
                        <option>7 Connecticut</option>
                        <option>8 Delaware</option>
                        <option>9 Florida</option>
                        <option>10 Georgia</option>
                        <option>11 Hawaii</option>
                        <option>12 Idaho</option>
                        <option>13 Illinois</option>
                        <option>14 Indiana</option>
                        <option>15 Iowa</option>
                        <option>16 Kansas</option>
                        <option>17 Kentucky</option>
                        <option>18 Louisiana</option>
                        <option>19 Maine</option>
                        <option>20 Maryland</option>
                        <option>21 Massachusetts</option>
                        <option>22 Michigan</option>
                        <option>23 Minnesota</option>
                        <option>24 Mississippi</option>
                        <option>25 Missouri</option>
                        <option>26 Montana</option>
                        <option>27 Nebraska</option>
                        <option>28 Nevada</option>
                        <option>29 New Hampshire</option>
                        <option>30 New Jersey</option>
                        <option>31 New Mexico</option>
                        <option>32 New York</option>
                        <option>33 North Carolina</option>
                        <option>34 North Dekota</option>
                        <option>35 Ohio</option>
                        <option>36 Oklahoma</option>
                        <option>37 Oregon</option>
                        <option>38 Pensylvania</option>
                        <option>39 Rhode Island</option>
                        <option>40 South Carolina</option>
                        <option>41 South Dakota</option>
                        <option>42 Tennessee</option>
                        <option>43 Texas</option>
                        <option>44 Utah</option>
                        <option>45 Vermont</option>
                        <option>46 Verginia</option>
                        <option>47 Washington</option>
                        <option>48 West Virginia</option>
                        <option>49 Wisconsin</option>
                        <option>50 Wyoming</option>
                    </select>
                </div>
                <div className='restaurant_field'>
                    <label htmlFor='restaurant_comments'><span className='comments_rest'>Comments</span></label>
                    <textarea id='restaurant_comments' name='restaurant-comments' placeholder='What shoule we know about this Restaurant?'></textarea>
                </div>
                <button type='submit' className='back-btn resForm'>Add</button>
            </form>
        )
    }
}
export default withRouter(AddRestaurant)