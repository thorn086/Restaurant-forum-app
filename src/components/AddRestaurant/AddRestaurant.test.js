import React from 'react';
import AddRestaurant from './AddRestaurant';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('AddRestaurant', () => {
    const addRestaurant= shallow(<AddRestaurant/>);
  
    it('renders properly', () => {
        expect(addRestaurant).toMatchSnapshot();
    });
});