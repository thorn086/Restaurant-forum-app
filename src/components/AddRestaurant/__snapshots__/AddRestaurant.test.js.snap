// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRestaurant renders properly 1`] = `ShallowWrapper {}`;
